packages/frontend/src/hooks/contracts/useController.ts

```diff
@@ -222,21 +222,48 @@ export const useController = () => {
     // return () => sub.unsubscribe()
   }, [web3, networkId, getIndex])
 
-  const getFundingForHalfHour = async () => {
-    let index
-    let mark
-    try {
-      index = await getIndex(1800)
-      mark = await getMark(1800)
-    } catch (error) {
+  // Tries to get funding for the longest period available based on Uniswap storage slots, optimistically 24hrs, worst case 30min
+  // TODO: get 24hr historical funding from the subgraph to have a value that isn't dynamic based on storage slots
+  const getDailyHistoricalFunding = async () => {
+    let index = new BigNumber(0)
+    let mark = new BigNumber(0)
+    let period = 24
+    let isError = false
+
+    //start by trying 24hr twap, if fails try dividing by 2 until 45min minimum, fall back to spot otherwise
+    for (; period >= 0.75; period = period / 2) {
+      try {
+        //convert period from hours to seconds
+        index = await getIndex(period * 3600)
+        mark = await getMark(period * 3600)
+      } catch (error) {
+        isError = true
+      }
+      if (isError === false) {
```

Comment:
don't you need to set isError to false if an error isnt caught (or inside the for loop in the try block)?

ie 24hrs doesnt work, but 12 does, this won't ever break out of loop right?